### ISSUE TITLE
make install_lua failing on Linux Mint 18.2

### ISSUE THREAD

**Issue by ghost** [hellerve/e#23]
I tried to compile the editor with lua support, and the `make install_lua` is failing. The `make`, and `make install` is running without problem.
[Make output](https://gist.github.com/CubeIX/fcc953690fb344874543458154300d2e)

**Comment by hellerve** [comment]
Which version of GCC did you compile it with?

**Comment by hellerve** [comment]
I just pushed a possible fix into the `issue_23` branch. Could you tell me whether that fixes it?

**Comment by ghost** [comment]
Yes your fix is working, but now the make is failing with a bunch of `undefined reference to lua_*` see [make output](https://gist.githubusercontent.com/CubeIX/7ce09ea4244abab534df6e549fd8dc37/raw/af0a677ea9b8c92e5c000ec6d35e4ce198466d44/issue%252323_luaerror)
gcc version:`5.4.0 20160609 (Ubuntu 5.4.0-6ubuntu1~16.04.4)`

**Comment by hellerve** [comment]
I assume that that’s an error on my side: I’m linking the vendored Lua, but not actually compiling it; this doesn’t make much of a difference on systems that have Lua installed, but I guess you don’t have it?

**Comment by ghost** [comment]
Yes I don't had Lua installed, but now I tried to compile first  the bundled lua soruce (with the `make generic` and with the `make linux` there is no difference, the editor's compilation is failing) ,and I tried install Lua5.3 before compiling, and it's failing as well. Any idea?

**Comment by hellerve** [comment]
I added Lua compilation to the Makefile and it works for me now if I uninstall Lua. Could you try pulling and compiling again (from `issue_23`)? Sorry for the hassle.

**Comment by ghost** [comment]
Well, I think you are in the right way to fix this bug. It's still failing, but I if I'm correct these errors is linking errors. [make output](https://gist.githubusercontent.com/CubeIX/99c55c984d458311f427f68d4c414d48/raw/0e46d3f256200f163da33201635002a693a1ea6a/issue%252323_luaerror_mod)

**Comment by hellerve** [comment]
Okay, so I needed to introduce two more options to the Makefile to get this to work. _sigh_

The magic incantation for me to get this to compile is: `make install_lua LUA_OPT=linux LUA_FLAGS="-lm -ldl"`. I also needed to install libreadline on Ubuntu, I’m not sure whether that’s also needed on Mint and whether we can compile Lua without readline support.

If you pull and try my incantation it might just work.
